app/retail/views.py

```diff
@@ -1001,6 +1001,27 @@ def activity(request):
 
     return TemplateResponse(request, 'activity.html', context)
 
+@ratelimit(key='ip', rate='30/m', method=ratelimit.UNSAFE, block=True)
+def create_status_update(request):
+    response = {}
+    if request.POST:
+        profile = request.user.profile
+        kwargs = {
+            'activity_type': 'status_update',
+            'metadata': {
+                'title': request.POST.get('data')
+            }
+        }
+        kwargs['profile'] = profile
+        try:
```

Comment:
rate limit pls. 

also we'll need to make sure ppl don't try to XSS here.

Reply:
CSRF is in place and there is rate limit decorator on top of the methid